src/cryptsetup/cryptsetup.c

```diff
@@ -1044,17 +1073,22 @@ static int attach_luks_or_plain_or_bitlk_by_pkcs11(
         for (;;) {
                 bool processed = false;
 
-                r = decrypt_pkcs11_key(
-                                name,
-                                friendly,
-                                uri,
-                                key_file, arg_keyfile_size, arg_keyfile_offset,
-                                key_data, key_data_size,
-                                until,
-                                arg_headless,
-                                &decrypted_key, &decrypted_key_size);
-                if (r >= 0)
-                        break;
+                if (use_libcryptsetup_plugin == true && arg_pkcs11_uri_auto) {
+                        r = attach_luks2_by_pkcs11(cd, name, friendly, until, arg_headless, flags);
```

Comment:
hm, you need to escape the loop here in some conditions, otherwise we'll wait for the pkcs11 device to pop up for good

Reply:
Confused again so in follow-up to https://github.com/systemd/systemd/pull/19797#discussion_r690358388 I most probably miss something about sd_event_wait()... How (where exactly) did you espace the loop in-before this PR?

Reply:
ignore me